src/main/telemetry/crsf.c

```diff
@@ -391,30 +391,31 @@ void crsfFrameFlightMode(sbuf_t *dst)
     // Acro is the default mode
     const char *flightMode = "ACRO";
 
-#if defined(USE_GPS)
-    if (!ARMING_FLAG(ARMED) && featureIsEnabled(FEATURE_GPS) && (!STATE(GPS_FIX) || !STATE(GPS_FIX_HOME))) {
-        flightMode = "WAIT"; // Waiting for GPS lock
-    } else
-#endif
-
     // Flight modes in decreasing order of importance
-    if (FLIGHT_MODE(FAILSAFE_MODE)) {
+    if (FLIGHT_MODE(FAILSAFE_MODE) || IS_RC_MODE_ACTIVE(BOXFAILSAFE)) {
         flightMode = "!FS!";
-    } else if (FLIGHT_MODE(GPS_RESCUE_MODE)) {
+    } else if (FLIGHT_MODE(GPS_RESCUE_MODE) || IS_RC_MODE_ACTIVE(BOXGPSRESCUE)) {
         flightMode = "RTH";
     } else if (FLIGHT_MODE(PASSTHRU_MODE)) {
         flightMode = "MANU";
     } else if (FLIGHT_MODE(ANGLE_MODE)) {
-        flightMode = "STAB";
+        flightMode = "ANGL";
     } else if (FLIGHT_MODE(HORIZON_MODE)) {
         flightMode = "HOR";
     } else if (airmodeIsEnabled()) {
         flightMode = "AIR";
     }
 
     sbufWriteString(dst, flightMode);
-    if (!ARMING_FLAG(ARMED)) {
-        sbufWriteU8(dst, '*');
+    if (!ARMING_FLAG(ARMED) && !FLIGHT_MODE(FAILSAFE_MODE)) {
+        // * - ready to arm
+        // ! - arming disabled
+        // ? - GPS fix not available
+        bool isGpsWaiting = false;
+#ifdef USE_GPS
+        isGpsWaiting = featureIsEnabled(FEATURE_GPS) && (!STATE(GPS_FIX) || !STATE(GPS_FIX_HOME));
+#endif
```

Comment:
I think that isGpsWaiting should be true when the FC has an arming block because it needs a Home fix, but doesn't have one.
A Home Fix requires a 3D fix (GPS_FIX) and that the minimum sat count is realised.
I think that the only situation where arming is blocked due to not having a Home fix is when the failsafe mode is set to GPS Rescue, AND you haven't chosen to take off without a Home fix, AND you don't have a 3D_FIX, AND you don't have the required locked satellite count.
The test for isGpsWaiting doesn't check these things correctly.
However, in that specific case, there is an arming block ARMING_DISABLED_GPS set in core.c around line 348.
Maybe we only need to test for that specific arming block, and apply '?' in that case.

Reply:
In the situation of GPS Rescue with no Home Point, it's impossible to arm.  With the current coding, the `?` symbol means both 'cannot arm *and* reason is no GPS Home point'. 
If arming is possible, and GPS Rescue is set, then user must have disabled the Home Point check for GPS Rescue.  Since no other GPS modes require a home point check, the `?` will not be shown.
Hence, with the current coding, `?` should take precedence over `!` since otherwise we will never see `?`.

If we want `?` to also mean 'no home fix', generally, then we would have to include an 'or' condition for `STATE(GPS_FIX_HOME)`, so that if arming was possible, but there was no home fix, because the user disabled the arming block in the CLI, they would then see `?`.  In this case the `?` would persist after arming, since it's not allowed to acquire a home point after arming.  It would equate to "I don't know where home is, so GPS Rescue cannot be initiated".

To me, that makes sense.  The `?` would mean:
- before arming : arming is blocked because you requested a home fix before arming, and there is no home fix; or, you don't care about a home point before arming, and we don't have one, so if you take off now, GPS Rescue will not work.
- after arming : you allowed arming without a home fix, and there was/is no home point, so I don't know where Home is, and GPS Rescue will not wor).

That makes sense to me.

Reply:
Leaving "?" character in above case in both cases disarmed/armed means we loose ability to know if aircraft is actually ARMED. From my point of view visual info that aircraft is armed (read BE CAREFUL) is more important than information I have no GPS home fix.

If '?" will be added only when disarmed it will serve as kind of warning "you can arm but be careful as GPS Rescue will not work"

IMHO priority   ! -> (? | *) -> no-char (armed) is better.
1. First we check for anything that pprevents arming. If there is FAILSAFE, RX LOSS, SMALL_ANGLE, ARM_SW it doesn't matter if GPS Home fix is available because you won't take off anyway and first you need to solve other issues. In this case "ACRO!"
2. Then if everything is okay ready to arm  we check if 
GPS connected & GPS MODE is set & ARMING_DISABLED_GPS is set & "Allow arming without fix" is set
So basically pilot decided he can arm without fix.
And we add '?' (disarmed-ready-to-arm-with-limitation) "ACRO?"
3. In any other case we add '*" (disarmed-ready-to-arm) "ACRO*"
4. When armed all characters are gone to have confirmation aircraft is actually armed. "ACRO"

Reply:
@JimB40 : 👍 ; maybe display ? even when GPS is blocking ARM (meaning more time is needed, everything other is fine)

Reply:
As I explained earlier, the detail in your point 2 is not needed. The flag that is set when gps rescue prevents arming is the only thing we need to check. The other things are already included in the gps rescue arming block Boolean.